core/trino-main/src/test/java/io/trino/execution/scheduler/TestingExchange.java

```diff
@@ -76,9 +77,19 @@ public Set<TestingExchangeSinkHandle> getFinishedSinkHandles()
     }
 
     @Override
-    public CompletableFuture<List<ExchangeSourceHandle>> getSourceHandles()
+    public ExchangeSourceHandleSource getSourceHandles()
```

Comment:
Maybe to avoid duplicated `source` in name (may make variable naming in usage confusing) this should be named `ExchangeSourceHandleProvider` or `ExchangeSourceHandleIterator`?

Reply:
The `ExchangeSourceHandleSource` name is consistent with `SplitSource`, though yeah, I agree, having the `Source` in the name twice makes it sound tautological. I don't have a strong opinion here, but somehow I'm more leaning towards keeping the naming consistent.

Reply:
No strong feeling also. Keep then.